src/modal/modal-manager.ts

```diff
@@ -68,19 +63,33 @@ export class ModalManager {
     }
   }
 
-  hideModal(): void {
-    if (this.modalElement) {
-      this.modalElement.modalHidden = true;
-      this.unregisterRootClickListener();
+  removeModal(): void {
+    const modal = this.getModal();
```

Comment:
how about only performing this action when this.modalElement exists? Otherwise, it will be added and then immediately removed, which seems unnecessary.